### ISSUE TITLE
GuildMemberJoinEvent is not fired on member rejoin after being in voice channel

### ISSUE THREAD

**Issue by GoldRenard** [DV8FromTheWorld/JDA#1169]
## General Troubleshooting

- [X] I have checked for similar issues.
- [X] I have updated to the latest JDA version.
- [X] I have checked the branches or the maintainers' PRs for upcoming bug fixes.

## Bug Report
When a member leaves the guild after being in a voice channel of this guild and rejoins, the `GuildMemberJoinEvent` will not be fired.

### Expected Behavior
The `GuildMemberJoinEvent` event is being fired.

### Reproduction Steps
1. Join voice channel;
2. Leave this guild (stay in vc);
3. Join this guild again;
4. Check if `GuildMemberJoinEvent` was fired for this member.

### Exception or Error
There is no any exception but enabling trace logging for `WebSocketClient` shows some interesting in this case:
```
n.d.j.internal.requests.WebSocketClient  : GUILD_MEMBER_ADD -> {/*...*/}
n.d.j.internal.requests.WebSocketClient  : Ignoring duplicate GUILD_MEMBER_ADD for user with id $USER_ID in guild $GUILD_ID
```

**Comment by MinnDevelopment** [comment]
Can someone confirm whether this is fixed by #1170 ?

**Comment by MinnDevelopment** [comment]
This should be fixed as of [4.0.0_80](https://ci.dv8tion.net/job/JDA/80/)

**Comment by GoldRenard** [comment]
I can confirm that. Fixed.
